src/utils/pgwire/src/pg_extended.rs

```diff
@@ -20,19 +20,47 @@ use regex::Regex;
 use crate::pg_field_descriptor::{PgFieldDescriptor, TypeOid};
 use crate::pg_protocol::cstr_to_str;
 
-fn replace_params(query_string: String, generic_params: &[usize], params: &[Bytes]) -> String {
+fn replace_params(query_string: String, generic_params: &[usize], params: &[String]) -> String {
     let mut tmp = query_string;
+    tmp.push(' ');
     for &i in generic_params.iter() {
```

Comment:
Use meaningful name like `for param in generic_params`. And I think .iter() is redudant.